File: SurTerrain/src/Models/Entete_facture.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package models;

import java.sql.Date;

/**
 *
 * @author ArwaBj
 */
public class Entete_facture {
   private int num_piece;
   private String type;
   private Date date_exp;
   private int tier;
   private int ref_facture;

    public Entete_facture(int num_piece, String type, Date date_exp, int tier,int ref_facture) {
        this.num_piece = num_piece;
        this.type = type;
        this.date_exp = date_exp;
        this.tier = tier;
        this.ref_facture = ref_facture;
    }

    public Entete_facture(Date date_exp, int tier, int ref_facture) {
        this.date_exp = date_exp;
        this.tier = tier;
        this.ref_facture = ref_facture;
    }

    public Entete_facture(String type, Date date_exp, int tier, int ref_facture) {
        this.type = type;
        this.date_exp = date_exp;
        this.tier = tier;
        this.ref_facture = ref_facture;
    }

    

    
   
   public Entete_facture(int ref_facture) {
        this.ref_facture = ref_facture;
    }


    

    

    public Entete_facture(Integer parseInt ) {
    }

   
   
   
    
   

    public int getNum_piece() {
        return num_piece;
    }

    public String getType() {
        return type;
    }

    public Date getDate_exp() {
        return date_exp;
    }

    public int getTier() {
        return tier;
    }

    public void setNum_piece(int num_piece) {
        this.num_piece = num_piece;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setDate_exp(Date date_exp) {
        this.date_exp = date_exp;
    }

    public void setTier(int tier) {
        this.tier = tier;
    }

    @Override
    public String toString() {
        return "Entete_facture{" + "num_piece=" + num_piece + ", type=" + type + ", date_exp=" + date_exp + ", Tier=" + tier + ", Ref_facture=" + ref_facture +'}';
    }

    public int getRef_facture() {
        return ref_facture;
    }

    public void setRef_facture(int ref_facture) {
        this.ref_facture = ref_facture;
    }
  
   
}

File: src/Entity/EnteteFacture.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * EnteteFacture
 *
 * @ORM\Table(name="entete_facture")
 * @ORM\Entity
 * @ORM\Entity(repositoryClass="App\Repository\EnteteFactureRepository")
 */
class EnteteFacture
{
    /**
     * @var int
     *
     * @ORM\Column(name="num_piece", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $numPiece;

    /**
     * @var string
     *
     * @ORM\Column(name="type", type="string", length=5, nullable=false)
     */
    private $type;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_exp", type="date", nullable=false)
     */
    private $dateExp;

    /**
     * @var int
     *
     * @ORM\Column(name="tier", type="integer", nullable=false)
     */
    private $tier;

    /**
     * @var int
     *
     * @ORM\Column(name="ref_facture", type="integer", nullable=false)
     */
    private $refFacture;

    public function getNumPiece(): ?int
    {
        return $this->numPiece;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getDateExp(): ?\DateTimeInterface
    {
        return $this->dateExp;
    }

    public function setDateExp(\DateTimeInterface $dateExp): self
    {
        $this->dateExp = $dateExp;

        return $this;
    }

    public function getTier(): ?int
    {
        return $this->tier;
    }

    public function setTier(int $tier): self
    {
        $this->tier = $tier;

        return $this;
    }

    public function getRefFacture(): ?int
    {
        return $this->refFacture;
    }

    public function setRefFacture(int $refFacture): self
    {
        $this->refFacture = $refFacture;

        return $this;
    }


}
